rest/sgcollect.go

```diff
@@ -146,12 +150,14 @@ func (c *sgCollectOptions) Validate() error {
 		}
 	}
 
+	// Customer number is required if uploading.
 	if c.Upload && c.Customer == "" {
 		return errors.New("customer must be set if upload is true")
 	}
 
-	if !c.Upload && c.UploadHost != "" {
-		return errors.New("upload must be set if upload_host is specified")
+	// If upload is false, and a custom upload_host is provided, ask them to set upload=true.
+	if c.UploadHost != defualtSGUploadHost && !c.Upload {
```

Comment:
Can this be simplified by moving 'setDefaults' to the end of Validate (or even have the caller of Validate invoke setDefaults?)

I think it's easier to reason about if 'Validate' is strictly validating the user input.